test/exchange_adapters/kucoin.test.ts

```diff
@@ -0,0 +1,246 @@
+          changePrice: '0.00251',
+          high: '0.107398',
+          low: '0.100304',
+          vol: '935258.2908',
+          volValue: '97271.9733359816',
+          last: '0.10551',
+          averagePrice: '0.10380346',
+          takerFeeRate: '0.001',
+          makerFeeRate: '0.001',
+          takerCoefficient: '1',
+          makerCoefficient: '1',
+        },
+        {
+          symbol: 'CELO-USDT',
+          symbolName: 'CELO-USDT',
+          buy: '0.7555',
+          sell: '0.7563',
+          changeRate: '0.0907',
+          changePrice: '0.0629',
+          high: '0.8281',
+          low: '0.6654',
+          vol: '1598294.854',
+          volValue: '1213224.94637127',
+          last: '0.7561',
+          averagePrice: '0.75703415',
+          takerFeeRate: '0.001',
+          makerFeeRate: '0.001',
+          takerCoefficient: '1',
+          makerCoefficient: '1',
+        },
+      ],
+    },
+  }
+
+  const inValidMockTickerJson = {
+    code: '200000',
+    data: {
+      time: undefined,
+      ticker: [
+        {
+          symbol: 'NKN-USDT',
+          symbolName: 'NKN-USDT',
+          buy: '0.105492',
+          sell: '0.105599',
+          changeRate: '0.0243',
+          changePrice: '0.00251',
+          high: '0.107398',
+          low: '0.100304',
+          vol: '935258.2908',
+          volValue: '97271.9733359816',
+          last: '0.10551',
+          averagePrice: '0.10380346',
+          takerFeeRate: '0.001',
+          makerFeeRate: '0.001',
+          takerCoefficient: '1',
+          makerCoefficient: '1',
+        },
+        {
+          symbol: 'CELO-USDT',
+          symbolName: 'CELO-USDT',
+          buy: undefined,
+          sell: undefined,
+          changeRate: '0.0907',
+          changePrice: '0.0629',
+          high: '0.8281',
+          low: '0.6654',
+          vol: undefined,
+          volValue: undefined,
+          last: undefined,
+          averagePrice: '0.75703415',
+          takerFeeRate: '0.001',
+          makerFeeRate: '0.001',
+          takerCoefficient: '1',
+          makerCoefficient: '1',
+        },
+      ],
+    },
+  }
+
+  describe('parseTicker', () => {
+    it('handles a response that matches the documentation', () => {
+      const ticker = kucoinAdapter.parseTicker(validMockTickerJson)
+      expect(ticker).toEqual({
+        source: Exchange.KUCOIN,
+        symbol: kucoinAdapter.standardPairSymbol,
+        ask: new BigNumber(0.7563),
+        baseVolume: new BigNumber(1598294.854),
+        bid: new BigNumber(0.7555),
+        lastPrice: new BigNumber(0.7561),
+        quoteVolume: new BigNumber(1213224.94637127),
+        timestamp: 1674501725001,
+      })
+    })
+
+    it('throws an error when a json field mapped to a required ticker field is missing or empty', () => {
+      expect(() => {
+        kucoinAdapter.parseTicker(inValidMockTickerJson)
+      }).toThrowError('timestamp, bid, ask, lastPrice, baseVolume not defined')
```

Comment:
should the error message be something in lines of timestamp or bid, etc.. because it could be that only 1 field is undefined

Reply:
based on the fields that are not defined or empty the error message is constructed so in this case all are in the error message because all are undefined